packages/tailwindcss/src/utilities.ts

```diff
@@ -5744,6 +5753,36 @@ export function createCssUtility(node: AtRule) {
               let value = node.value.replace(/-\*.*$/g, '') as `--${string}`
               storage[fn.value].themeKeys.add(value)
             }
+
+            // Validate bare value data types
+            else if (
+              node.kind === 'word' &&
+              !(node.value[0] === '[' && node.value[node.value.length - 1] === ']') && // Ignore arbitrary values
+              !BARE_VALUE_DATA_TYPES.includes(node.value)
+            ) {
+              console.warn(
+                `Unsupported bare value data type: "${node.value}".\nOnly valid data types are: ${BARE_VALUE_DATA_TYPES.map((x) => `"${x}"`).join(', ')}\n`,
```

Comment:
@adamwathan do you want to make any changes to this warning?

Output looks something like this right now:
~~~
Unsupported bare value data type: "color".
Only valid data types are: "number", "integer", "ratio", "percentage"

```css
--value([color],color)
                ^^^^^
```
~~~

Reply:
My 5 ct: I don't think the term "bare value" is something that people of that API will be familiar with but maybe we just call it "value data type" (since it's the `--value(…)` function)? 

````
Unsupported value data type: "color".
Only valid data types are: "number", "integer", "ratio", "percentage".

```css
--value([color],color)
                ^^^^^
```
````

The rest I think is fine personally but I'd add a dot at the end of the second sentence

Reply:
I think people writing plugins like that will be more familiar with the term (or at least should be because they will have to know the difference between `color` and `[color]`). 

Main reason I used it is because that's what we use in the docs for it: https://tailwindcss.com/docs/adding-custom-styles#bare-values